toolchain/sem_ir/typed_insts.h

```diff
@@ -396,6 +397,9 @@ struct BoundMethod {
   // `self`, or whose address will be used to initialize `self` for an `addr
   // self` parameter.
   InstId object_id;
+  // The instruction from which to find the function being bound to an object.
+  // The actual function (e.g. FunctionType) comes from the this instruction's
+  // type id.
```

Comment:
Looking at `member_access.cpp`, `BoundMethod` will only be created for `FunctionType`. I'm not sure I'd call that "the actual function"; to me, that's either the ID here, or the `Function` object which contains the function implementation. That is, we're not using this to find the bound function, it _is_ the bound function.

What do you think of:

```suggestion
  // The bound function, whose type_id is always a `FunctionType`.
```

Reply:
Sidenote, I think this could maybe be renamed to `function_decl_id` if you care -- we typically use `function_id` for `FunctionId`, but that's not the case here. I do think though that this will match the decl concept (although it may be an `ImportRef` instead of a `FunctionDecl` sometimes? I'm not strictly verifying this)

Reply:
> Looking at member_access.cpp, BoundMethod will only be created for FunctionType. 

I was wondering if it may eventually also be used with SpecificFunctionType? Until/unless that's the case, this looks good yeah. Or could we say "The declaration of the function being bound"?

> Sidenote, I think this could maybe be renamed to function_decl_id if you care

I like that, I will do that separately.

Reply:
Richard says a specific function is formed around the bound method when it's being called, after member access. i.e., SpecificFunctionType -> BoundMethod -> FunctionType

Reply:
Ah okay, thanks :) I will apply the change as is.